### DISCUSSION TITLE
Value of Textarea after reset not ""

### DISCUSSION THREAD

**Discussion by andy-schwedler** [neuefische/questions#454] [Web] [answered]
Hi there, 

I have an input `type=textarea`, which get´s reset after `submit`.

Additionally, I have a `counter` that counts `event.target.length` for the textare input. 

Unfortunately, after submit the counter does not reset to 0.

[PR](https://github.com/andy-schwedler/capstone-project/pull/37/commits/e0bd6e5734759e7a87e0e47b430d742307e0977f) here

here´s the handler function

```js
  function handleCounter(event) {
    setCounter(event.target.value.length);
  }
```

expected behavior:

```js
  function handleCounter(event) {
    if (!event.target.value) {
      setCounter(0);
    }
    setCounter(event.target.value.length);
  }```

**Comment by andy-schwedler** [comment, marked as answer]
In this [PR](https://github.com/andy-schwedler/capstone-project/pull/37/commits/e0bd6e5734759e7a87e0e47b430d742307e0977f)
```js
<StyledCreateForm aria-label="create card form" onSubmit={()=>{onAddCreateCard; setCounter(0)}}> 
```

**Comment by MerleMak** [comment]
So I found this in you _app.js: 
``` js
event.target.reset();
event.target.headline.focus(); 
```
I think you could extract it from there and add it to a handleSubmit function inside of your form component. Like so:
``` js
function handleSubmit(event){
    onAddCreateCard(event)
    event.target.reset();
    event.target.headline.focus();
    setCounter(0)
  }
  ```
  This way you are making sure to add the event to `onAddCreateCard` and its also more readable. 
  naturally you shouldn't forget to change 
  ``` js
   <StyledCreateForm aria-label="create card form" onSubmit={handleSubmit}>
  ```

**Reply by andy-schwedler** [reply]
Most elegant answer!
